src/services/tiles.ts

```diff
@@ -163,21 +230,55 @@ const tilesInfo: Record<
     properties: ['id'],
     sourceLayer: 'zoneDP',
   },
+  buildings: {
+    source: 'database',
+    table: 'pre-table-buildings',
+    minZoom: minZoomData,
+    options: {
+      maxZoom,
+    },
+    properties: [
+      'id',
+      'nb_logements',
+      'annee_construction',
+      'type_usage',
+      'energie_utilisee',
+      'type_chauffage',
+      'addr_label',
+      'dpe_energie',
+      'dpe_ges',
+    ],
+    sourceLayer: 'buildings',
+  },
 };
 
+const promiseGetters: Promise<unknown>[] = [];
 Object.entries(tilesInfo).forEach(
   ([type, { source, table, options, properties }]) => {
-    debug && console.info(`Indexing tiles for ${type} with ${table}...`);
-    const getter =
-      source === 'airtable'
-        ? getObjectIndexFromAirtable
-        : getObjectIndexFromDatabase;
-    getter(table, options, properties).then((result) => {
-      allTiles[type as DataType] = result;
-      debug && console.info(`Indexing tiles for ${type} with ${table} done`);
-    });
+    if (!allTiles[type as DataType]) {
+      debug && console.info(`Indexing tiles for ${type} with ${table}...`);
+      const getter =
+        source === 'airtable'
+          ? getObjectIndexFromAirtable
+          : getObjectIndexFromDatabase;
+      promiseGetters.push(
+        getter(table, options, properties).then((result) => {
+          allTiles[type as DataType] = result;
+          debug &&
+            console.info(`Indexing tiles for ${type} with ${table} done`);
+        })
+      );
+    } else {
+      debug &&
+        console.info(
+          `Indexing tiles for ${type} with ${table} already completed`
+        );
+    }
   }
 );
+Promise.all(promiseGetters).then(
```

Comment:
Why ?

Reply:
It's just for a debug log when all indexing is done. 
I find it useful, but if you prefer we can delete it.

Reply:
okay for me